drivers/SmartThings/zwave-switch/src/aeotec-smart-switch/init.lua

```diff
@@ -42,12 +104,49 @@ local function on_off_factory(onOff)
   end
 end
 
+local function set_color(driver, device, command)
+  local r, g, b = utils.hsl_to_rgb(command.args.color.hue, command.args.color.saturation, command.args.color.lightness)
+
+  local set = SwitchColor:Set({
+    color_components = {
+      { color_component_id=SwitchColor.color_component_id.RED, value=r },
+      { color_component_id=SwitchColor.color_component_id.GREEN, value=g },
+      { color_component_id=SwitchColor.color_component_id.BLUE, value=b },
+    }
+  })
+  device:send(set)
+
+  local query_color = function()
+    device:send(
+      SwitchColor:Get({ color_component_id=SwitchColor.color_component_id.RED }),
+      command.component
+    )
+  end
+
+  device.thread:call_with_delay(constants.DEFAULT_GET_STATUS_DELAY, query_color)
+end
```

Comment:
can I ask why this was redefined over the defaults?

Reply:
I added the handler because if it's not set, the color setting for the LED doesn't work. 
How can I set the default handler for colorControl in the driver? 
Or does it inherit from the parent driver? But if it inherits from the parent driver, why doesn't it work?

Reply:
it should be inherited from the parent driver, but the default uses the warm_white and cold_white params (set to 0). Perhaps that's why it doesn't work with your device.

Reply:
Okay, so can I leave my set_color handler in the sub-driver?

Reply:
Sure, though I _think_ the Z-Wave spec says unexpected extra fields should just be ignored, rather than cause the command to fail, so worth keeping that in mind.